Ticket-booking-spring-app/src/main/java/com/example/demo/service/seat_availability/SeatAvailabilityServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.example.demo.service.seat_availability;
+
+import com.example.demo.dao.SeatAvailabilityRepository;
+import com.example.demo.entity.Screening;
+import com.example.demo.entity.Seat;
+import com.example.demo.entity.SeatAvailability;
+import com.example.demo.exception.DataBaseException;
+import com.example.demo.exception.ScreeningException;
+import com.example.demo.service.screening.ScreeningService;
+import com.example.demo.service.seat.SeatService;
+import com.example.demo.service.seat_availability.SeatAvailabilityService;
+import com.example.demo.util.SeatAvailabilityResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class SeatAvailabilityServiceImpl implements SeatAvailabilityService {
+    private SeatAvailabilityRepository seatAvailabilityRepository;
+    private SeatService seatService;
+    private ScreeningService screeningService;
+
+    @Autowired
+    public SeatAvailabilityServiceImpl(
+            SeatAvailabilityRepository theSeatAvailabilityRepository,
+            SeatService theSeatService,
+            ScreeningService theScreeningService) {
+        seatAvailabilityRepository = theSeatAvailabilityRepository;
+        seatService = theSeatService;
+        screeningService = theScreeningService;
+    }
+
+    @Override
+    public Map<Integer, Boolean> seatToAvailabilityByScreeningId(int screeningId) {
+        List<SeatAvailability> availabilities = seatAvailabilityRepository.
+                findByScreeningId(screeningId);
+        Map<Integer, Boolean> result = new HashMap<>();
+        for (SeatAvailability availability : availabilities) {
```

Comment:
Spróbujesz przerobić to na stream? :)

Reply:
List<SeatAvailability> availabilities = seatAvailabilityRepository.
        findByScreeningId(screeningId);
return availabilities.stream()
        .collect(Collectors.toMap(SeatAvailability::getSeatId, SeatAvailability::getAvailable));
       
Coś takiego? :)